### PR TITLE
fix: pipeline 3000 creating configs

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Creation was broken because of permission checks

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/scenes/pipeline/utils.tsx [hunk 1]
```diff
@@ -369,26 +372,27 @@ export function pipelinePluginBackedNodeMenuCommonItems(
...
-                      disabledReason: canConfigurePlugins ? undefined : 'You do not have permission to delete.',
+                      disabledReason: notAllowedReasonByStageAndOperationType[node.stage]['edit_without_enable'],
                   },
               ]
             : []),
     ]
 }
 
-export function checkPermissions(stage: PipelineStage, togglingToEnabledOrNew: boolean): boolean {
-    const { canConfigurePlugins, canEnableNewDestinations } = useValues(pipelineLogic)
```
Comment (tiina303) [inline frontend/src/scenes/pipeline/utils.tsx:380]: I suspect this is the reason why updates started failing, we can't call a function with using from a logic

File: frontend/src/scenes/pipeline/pipelineLogic.tsx [hunk 2]
```diff
@@ -59,11 +62,52 @@ export const pipelineLogic = kea<pipelineLogicType>([
         // it's better to add the permission checks everywhere now
         canGloballyManagePlugins: [(s) => [s.user], (user) => canGloballyManagePlugins(user?.organization)],
         canConfigurePlugins: [(s) => [s.user], (user) => canConfigurePlugins(user?.organization)],
-        canEnableNewDestinations: [
-            (s) => [s.user, s.hasAvailableFeature],
-            (user, hasAvailableFeature) =>
-                user?.is_impersonated ||
-                (canConfigurePlugins(user?.organization) && hasAvailableFeature(AvailableFeature.DATA_PIPELINES)),
+        notAllowedReasonByStageAndOperationType: [
```
Comment (tiina303) [inline frontend/src/scenes/pipeline/pipelineLogic.tsx:65]: not sure this is the best way to do this, but having a consistent disabled reason / error is nice, later other things might be allowed / disallowed and that can be changed here

Comment (benjackwhite) [reply frontend/src/scenes/pipeline/pipelineLogic.tsx:65]: I found this really hard to parse overall as a concept tbh. I don't fully get what it is trying to achieve tbh. 

My gut tells me this shouldn't be abstracted here but rather implemented in each logic for the different kinds of thing. Reason being that the benefits of abstracting it here seem very light in comparison to the high complexity of passing this object around everywhere.

For example the `NewButton` is made relatively complex as it is trying to be super abstract when in actual fact the logic to decide whether it should be shown could easily belong to the more specific logic like `pipelineDestinationsLogic` and then the `NewButton` can remain simple and not need this passed object. 

Might be worth an in person sync if this doesn't make sense 👍

**Review by benjackwhite — CHANGES_REQUESTED** [review]
